fix: HAIP interoperability for x509_hash

ZAnsaroudi opened the pull request:
This PR resolves #806. 

Particularly: 
- Highlights the differences in `client_id` usage, especially when required for signature and correctness checks.
- Updates `client_metadata` to include the required parameters when `x509_hash` is used.

ZAnsaroudi committed:
fix: HAIP interoperability for x509_hash

ZAnsaroudi committed:
fix: small update to x5c

ZAnsaroudi committed:
fix: small update to error description (prefix requirements)

peppelinux commented on docs/en/relying-party-metadata.rst:
```diff
@@ -5,6 +5,9 @@ Relying Party Metadata
 The *openid_credential_verifier* metadata MUST contain the following parameters (:ref:`test-plans-remote-presentation`).
 
+.. note::
+  The metadata below applies only when the ``client_id`` uses the ``openid_federation`` prefix. For ``x509_hash``, the metadata is instead conveyed in the ``client_metadata`` parameter. 
```
```suggestion
The *openid_credential_verifier* metadata MUST contain the *client_metadata*, as included in the parameters shown below. (:ref:`test-plans-remote-presentation`). Please note that *openid_credential_verifier* is a OpenID Federation specific metadata used for OpenID4VP and it is used only when the Relying Party's ``client_id`` is set with ``openid_federation``. When the Relying Party's ``client_id`` parameter is set with ``x509_hash``, the metadata is instead conveyed in the ``client_metadata`` parameter provided within the request.
```

ZAnsaroudi committed:
Update docs/en/relying-party-metadata.rst

Co-authored-by: Giuseppe De Marco <demarcog83@gmail.com>

peppelinux commented on docs/en/remote-flow.rst:
```diff
@@ -38,13 +38,15 @@ A High-Level description of the remote flow, from the User's perspective, is giv
-  6. *POST Authorization Response*: the Wallet Instance presents the requested information to the Relying Party, along with the Wallet Attestation if requested.
-  7. *RP Checks*: The Relying Party validates the presented Credentials by verifying the trust with their Issuers and checks the Wallet Attestation to ensure the Wallet Provider is trusted.
-  8. *Relying Party Response*: the Wallet Instance informs the User about the successful authentication with the Relying Party, and the User continues the navigation.
+    c. evaluates the requested Digital Credentials and checks the eligibility of the Relying Party in asking for these by applying the policies related to that specific Relying Party, obtained with the Trust Chain (:ref:`WP_087 <wallet-credential-presentation-testcases>`).
```
that's ok for this release. For the next milestone we'll implement RP's policies according to ARF's STS, as in review in weBuild LSP, see: https://github.com/webuild-consortium/wp4-trust-group/pull/12/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R125

peppelinux commented on docs/en/remote-flow.rst:
```diff
@@ -403,7 +405,11 @@ The JWT header parameters are described below:
     - Key ID of the public key needed to verify the JWT signature, as defined in [:rfc:`7517`]. REQUIRED when ``trust_chain`` is used.
   * - **trust_chain**
     - Sequence of Entity Statements that composes the Trust Chain related to the Relying Party, as defined in `OID-FED`_ Section 4.3 *Trust Chain Header Parameter*.
```
```suggestion
    - CONDITIONAL. REQUIRED when the ``client_id`` prefix used in the request is set with ``openid-federation``. It is a sequence of Entity Statements that composes the Trust Chain related to the Relying Party, as defined in `OID-FED`_ Section 4.3 *Trust Chain Header Parameter*.
```

peppelinux commented on docs/en/remote-flow.rst:
```diff
@@ -416,6 +422,14 @@ The JWT payload parameters are described herein:
+        - **encrypted_response_enc_values_supported**. JSON array listing the supported JWE ``enc`` algorithms for encrypted Authorization Responses in ``direct_post.jwt``.
+        - **jwks**. JSON Web Key Set used by the Wallet Instance for encrypting the Authorization Response or for key agreement. Keys contained in this set are request-specific and identified by their ``kid`` value.
+        - **request_uris**. JSON Array of pre-registered request_uri endpoints where the Wallet Instance may obtain a new Request Object. URLs MUST use the ``https`` scheme.
+        - **response_uris**. JSON Array of endpoints to which the Wallet Instance sends the Authorization Response using ``direct_post.jwt``. URLs MUST use the ``https`` scheme.
```
```suggestion
```

peppelinux commented on docs/en/remote-flow.rst:
```diff
@@ -403,7 +405,11 @@ The JWT header parameters are described below:
   * - **trust_chain**
     - Sequence of Entity Statements that composes the Trust Chain related to the Relying Party, as defined in `OID-FED`_ Section 4.3 *Trust Chain Header Parameter*.
+  * - **x5c**
+    - CONDITIONAL. REQUIRED when ``client_id`` uses the ``x509_hash`` prefix; otherwise ``kid`` with ``jwks`` is used. Contains the Relying Party’s leaf X.509 certificate (and optionally intermediate certificates), used to verify the JWT signature with the public key in the Relying Party’s certificate as defined in :rfc:`7515`. The Relying Party’s certificate in ``x5c`` MUST assert identity infor ...
```
```suggestion
    - CONDITIONAL. REQUIRED when ``client_id`` uses the ``x509_hash`` prefix; otherwise ``kid`` with ``jwks`` is used. Contains the Relying Party’s leaf X.509 certificate (and optionally intermediate certificates), used to verify the JWT signature with the public key in the Relying Party’s certificate as defined in :rfc:`7515`. The Relying Party’s certificate in ``x5c`` MUST assert Relying Party identity information. When the Relying Party's X.509 certificate contains X.509 SAN URI parameters, the endpoints used in the presentation request MUST be validated against X.509 SAN URI parameters value, such as ``response_uri`` and ``redirect_uri``. See Section 5 of `OPENID4VC-HAIP`_.
```

ZAnsaroudi committed:
Update docs/en/remote-flow.rst

Co-authored-by: Giuseppe De Marco <demarcog83@gmail.com>

ZAnsaroudi committed:
Update docs/en/remote-flow.rst

- CONDITIONAL. REQUIRED when ``client_id`` uses the ``x509_hash`` prefix; otherwise ``kid`` with ``jwks`` is used. Contains the Relying Party’s leaf X.509 certificate (and optionally intermediate certificates), used to verify the JWT signature with the public key in the Relying Party’s certificate as defined in :rfc:`7515`. The Relying Party’s certificate in  ``x5c`` MUST assert the Relying Party identity information sufficient to bind the network endpoints referenced by the presentation flow. In particular, the endpoints used in the Authorization Request and Authorization Response (e.g., ``response_uri``, ``redirect_uri``) MUST correspond to identity information contained in the Relying Party’s certificate (for example, a URI-type SAN for full-URI matching or a DNSName SAN for host-name matching).

Co-authored-by: Giuseppe De Marco <demarcog83@gmail.com>

ZAnsaroudi committed:
fix: small update to x5c

ZAnsaroudi committed:
Update docs/en/remote-flow.rst

Co-authored-by: Giuseppe De Marco <demarcog83@gmail.com>

peppelinux committed:
chore: openid4vp HAIP x509 hash alignemnt italian translation

peppelinux commented on docs/en/relying-party-metadata.rst:
```diff
@@ -3,7 +3,7 @@
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The *openid_credential_verifier* metadata MUST contain the following parameters (:ref:`test-plans-remote-presentation`).
+The *openid_credential_verifier* metadata MUST contain the *client_metadata*, as included in the parameters shown below. (:ref:`test-plans-remote-presentation`). Please note that *openid_credential_verifier* is a OpenID Federation specific metadata used for OpenID4VP and it is used only when the Relying Party's ``client_id`` is set with ``openid_federation``. When the Relying Party's ``client_id` ...
```
```suggestion
The *openid_credential_verifier* metadata MUST contain the *client_metadata*, as included in the parameters shown below. (:ref:`test-plans-remote-presentation`). Please note that *openid_credential_verifier* is a OpenID Federation specific metadata used for OpenID4VP and it is used when the Relying Party's ``client_id`` is set with ``openid_federation``. When the Relying Party's ``client_id`` parameter is set with ``x509_hash``, the metadata is instead conveyed in the ``client_metadata`` parameter provided within the request.
```

peppelinux committed:
Apply suggestions from code review

peppelinux committed:
fix: RST linting

peppelinux committed:
Merge branch 'haip_rf_interoperability' of https://github.com/ZAnsaroudi/eudi-wallet-it-docs into haip_rf_interoperability
